Nome: Igor Lucas dos Santos Braz

Matricula: 3865

In [44]:
import pandas as pd
import numpy as np

In [45]:
from google.colab import drive
drive.mount('/content/drive')

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


1. Protocolos IA171 e IE666

In [46]:
ia171 = pd.read_csv('/content/atraso_ia171.csv', index_col=False, header=None, squeeze=True)

ia171.describe()

count    100.000000
mean     500.230600
std       23.901383
min      424.770000
25%      483.540000
50%      503.905000
75%      516.222500
max      558.330000
Name: 0, dtype: float64

In [47]:
ie666=pd.read_csv('/content/atraso_ie666.csv', index_col=False, header=None, squeeze=True)

ie666.describe()

count    100.000000
mean     186.781500
std       53.278042
min       54.960000
25%      154.040000
50%      178.895000
75%      230.262500
max      317.690000
Name: 0, dtype: float64

Fazendo as hipóteses, temos que a hipótese nula é => média(ia171)=média(ie666). Já a hipótese alternativo = média(ia171) != média(ie666)

Com isso, o teste bilateral, a diferença entre as médias pode ser para cima ou para baixo

In [74]:
media = ia171.mean()
desvio = ia171.std()
tam = len(ia171)
desvio_erro = desvio/np.sqrt(tam)

confianca = (media - (1.95*desvio_erro)) - (media +(1.95*desvio_erro))

print("media ia171: %.3f" % media)
print("Intervalo: [%.3f - %.3f]" % ((media - (1.96*desvio_erro)), (media +(1.96*desvio_erro))))

media ia171: 500.231
Intervalo: [495.546 - 504.915]


Agora fazendo o intervalo de confiança para ie666

In [75]:
media = ie666.mean()
desvio = ie666.std()
tam = len(ie666)
desvio_erro = desvio/np.sqrt(tam)

confianca = (media - (1.95*desvio_erro)) - (media +(1.95*desvio_erro))

print("media ie666: %.3f" % media)
print("Intervalo: [%.3f - %.3f]" % ((media - (1.96*desvio_erro)), (media +(1.96*desvio_erro))))

media ie666: 186.782
Intervalo: [176.339 - 197.224]


Com base nos intervalos de confiança já da para ter uma noção da diferença dos protocolos.
Com isso: 

In [76]:
media1 = ia171.mean()
media2 = ie666.mean()
desvio1 = ia171.std()
desvio2 = ie666.std()
tam1 = len(ia171)
tam2 = len(ie666)
desvio_erro_dif=np.sqrt((np.power(desvio2,2)/tam2) + (np.power(desvio1,2)/tam1))
dif_media=(media2 - media1)

print("Diferenças: %.3f" % dif_media)
print("Erro padrão da diferença: %.3f " % desvio_erro_dif)
print("As diferenças de média está %.3f vezes o erro padrao da média" % (dif_media/desvio_erro_dif))


Diferenças: -313.449
Erro padrão da diferença: 5.839 
As diferenças de média está -53.679 vezes o erro padrao da média


Como a diferença entre as médias é maior que 50 vezes o erro padrão da diferença das médias, é extramamente improvável que a hipótese nula se confirme. Nesse caso, podemos escolher o protocolo com menor média (no caso, ie666).

Mesmo assim, vamos calcular o P-Value considerando a distribuição normal, multiplicando por 2 por ser um teste bilateral.

Para isso, vamos assumir um alpha=5% (0,05). Então, caso p-value < 0,05, podemos rejeitara hipótese nula.


In [79]:
alpha=0.05
p_value_normal = ((1-st.norm.cdf(abs(2)))*2)
print("(Normal) valor-p= %.3f " % p_value_normal)
if(p_value_normal < alpha):
  print("Hipótese nula pode ser rejeitada")
else:
  print("Não podemos rejeitar a hipótese nula")

(Normal) valor-p= 0.046 
Hipótese nula pode ser rejeitada


In [81]:
# t-test é usado quando:
# a) Não sabemos a variancia da populacão
# b) Tamanho da amostra é pequeno (n < 30)
graus_liberdade = len(ie666) + len(ia171) - 2
p_value_t = (1-st.t.cdf(abs(desvio_erro_dif),graus_liberdade))*2
print("(T-Test) valor-p= %.3f " % p_value_t)

(T-Test) valor-p= 0.000 


A diferença entre as médias é 50 vezes maior que o erro padrõa da diferença das médias, com isso é improvável que a hipotese nula seja verdade. Portanto, escolhemos o protocolo com menor média(ie666).
Com isso, o protocolo com menor faixa do intervalo de confiança(ia666) deverá ser escolhido por ser o melhor.

**2. CombNar e BitComb **

In [60]:
combinar=pd.read_csv("/content/tempo_combinar.csv", index_col=False, header=None, squeeze=True)

combinar.describe()

count    100.000000
mean       9.540900
std        3.956263
min       -2.150000
25%        7.682500
50%        9.250000
75%       12.407500
max       18.400000
Name: 0, dtype: float64

In [61]:
bitcomb=pd.read_csv("/content/tempo_bitcomb.csv", index_col=False, header=None, squeeze=True)

bitcomb.describe()

count    100.000000
mean       8.743500
std        3.600244
min       -0.070000
25%        6.100000
50%        8.640000
75%       11.382500
max       17.200000
Name: 0, dtype: float64


Fazendo as hipóteses, temos que a hipótese nula é => média(Combinar)=média(Bitcomb). Já a hipótese alternativo = média(combinar) > média(bitcomb), considerando que um tempo maior é pior.

Teste unilateral pela direita:
Reijatando A HIPOTESE NULA SE A PROBABILIDADE DE ENCONTRAR UMA DIFERENÇA ENTRE AS MÉDIAS,CONFORME OBSERVADO NOS DADOS, SE SITUAR DENTRO DA REGIÃO CRITICA. SE CONSIDERARMOS UM NÍVEL DE SIGNIFICANCIA DE 5%, VAMOS REJEITAR A HIPOTESE NULA SE Z-SCORE > 1.96 OU SE P-VALUE < 0.05.



In [62]:
media = combinar.mean()
desvio = combinar.std()
tam = len(combinar)
desvio_erro = desvio/np.sqrt(tam)

print("Erro padrao: %.3f" % desvio_erro)
print("Intervalo: [%.3f - %.3f]" % ((media - (1.96*desvio_erro)), (media +(1.96*desvio_erro))))

Erro padrao: 0.396
Intervalo: [8.765 - 10.316]


agora BitComb

In [64]:
mediab = bitcomb.mean()
desviob = bitcomb.std()
tamb = len(bitcomb)
desvio_errob = desviob/np.sqrt(tamb)



print("Erro padrao: %.3f" % desvio_errob)
print("Intervalo: [%.3f - %.3f]" % ((mediab - (1.96*desvio_errob)), (mediab +(1.96*desvio_errob))))

Erro: 0.360
Intervalo: [8.038 - 9.449]


Agora fazendo as diferenças de médias:

In [70]:
media1 = combinar.mean()
media2 = bitcomb.mean()
desvio1 = combinar.std()
desvio2 = bitcomb.std()
tam1 = len(combinar)
tam2 = len(bitcomb)
desvio_erro_dif=np.sqrt((np.power(desvio2,2)/tam2) + (np.power(desvio1,2)/tam1))
dif_media=(media1 - media2)

print("Diferenças: %.3f" % dif_media)
print("Erro padrão da diferença: %.3f " % desvio_erro_dif)
print("As diferenças de média está %.3f vezes o erro padrao da média" % (dif_media/desvio_erro_dif))


Diferenças: 0.797
Erro padrão da diferença: 0.535 
As diferenças de média está 1.491 vezes o erro padrao da média


Vamos calcular o P-Value para a diferença entre as médias em termos do erro padrão. Nesse caso, por ser unilateral, não vamos multiplicar por 2. Outra diferença é que, como é unilateral para a esquerda, não fazemos (1 - CDF), pois a CDF já indica probabilidade cumulativa que vai da esquerda para direita. Porém, nesse caso temos que pegar o valor negativo.

Também vamos calcular o P-Value em considerando a distribuição normal e a distribuição T-Student.

In [72]:
alpha=0.05
p_value_normal = (st.norm.cdf(desvio_erro_dif))
print("(Normal) valor-p= %.3f " % p_value_normal)
if(p_value_normal < alpha):
  print("Hipótese nula pode ser rejeitada")
else:
  print("Não podemos rejeitar a hipótese nula")

(Normal) valor-p= 0.704 
Não podemos rejeitar a hipótese nula


In [73]:
graus_liberdade = len(combinar) + len(bitcomb) - 2
p_value_t = (st.t.cdf((-1)*desvio_erro_dif,graus_liberdade))
print("(T-Test) valor-p= %.3f " % p_value_t)

(T-Test) valor-p= 0.297 


Independente do tipo de teste (bilateral ou unilateral), o cálculo do valor-p é do mesmo jeito. O que muda é a rejeição ou não da H0. 

Os dois intervalos de confiança se sobrepões, e a diferença das médias é de apenas 1.491 desvios padrões. A probabilidade de se observar esses dados caso a H0 for verdadeira (M(combinar) <= M(bitcomb)) é de 0.74 (valor-p). Portanto, não podemo rejeitar a H0. Mas veja que isso não garante que H0 seja verdadeira. Além disso, H0 também considera a igualdade entre as duas.